tests/test_model_evaluation.py

```diff
@@ -72,3 +85,305 @@ def test_eval_function_calls(eval_func):
+
+        # get results from each evaluator in the list
+        map(lambda ev: ev.get_results(), evals)
+
+    # results of the evals passed to the model_evaluation function
+    results_1 = all_results(evaluators_1)
+    results_2 = all_results(evaluators_2)
+
+    # results of evals returned by the model_evaluation function
+    results_1_ = all_results(evaluators_1_)
+    results_2_ = all_results(evaluators_2_)
+
+    assert results_1 == results_2
+    assert results_1_ == results_2_
+
+
+class _MockLinearEstimator(BaseEstimator):
+    def __init__(self, coefs):
+        coefs = np.array(coefs)
+        assert len(coefs.shape) == 1
+        self._coefs = coefs
+
+    def fit(self, X, y):
+        pass
+
+    def predict(self, X):
+        y_pred = self._coefs @ X
+        return y_pred
+
+
+def test_bootstrap_samples_from_eval_distribution(n_repeats=10, seed=42):
+    """Test that true mean is in 95%CI of bootstrap samples.
+
+    If there is a very probability that it's not, this test fails.
+
+    This test simply repeats _test_bootstrap_samples_from_eval_distribution
+    and fails if it fails 100% of the time; chance of false failure is 0.05**(n_repeats).
+
+    The default of 10 repeats puts us at a 1:1e14 chance of false failure.
+
+    This is obviously at the expense of allowing more false passes.
+    """
+    # generate a sequence of random seeds
+    seeds = np.random.default_rng(seed).integers(10000, size=n_repeats)
+    logger.info(f"seeds {seeds}")
+
+    within_bound_list = [
+        _test_bootstrap_samples_from_eval_distribution(random_state)
+        for random_state in seeds
+    ]
+
+    assert np.any(within_bound_list)
+
+
+def _test_bootstrap_samples_from_eval_distribution(random_state):
+    """Test that true mean is in 95%CI of bootstrap samples.
+
+    This is a sanity test of bootstrapping from an Evaluator
+    that generates evaluations from a known distribution;
+    it does not make meaningful use of an estimator.
+
+    This is deterministic, assuming proper use of random seeds
+    (so won't break unless a dependent use of rngs is changed,
+    in which case there's a 5% chance of a false negative,
+    in which case change the seed?)
+    """
+    estimator = _MockEstimator()
+
+    X, y = pd.DataFrame(np.ones((100, 2))), pd.Series(np.ones(100))
+    mean = 0
+    stdev = 1
+
+    def sample_from_normal(rng):
+        return rng.normal(mean, stdev)
+
+    evaluator = _MockRandomEvaluator(sample_from_normal)
+    n_bootstrap_replications = 30
+
+    # seed from which to generate a sequence of random seeds
+    [evaluator_] = bootstrap_model(
+        estimator,
+        X,
+        y,
+        [evaluator],
+        replications=n_bootstrap_replications,
+        random_state=random_state,
+    )
+    results = evaluator_.get_results()
+    bs_mean = np.mean(results)
+
+    # with 95% probability,
+    # mean in bs_mean +- bound.
+    bound = 1.96 * (stdev / (np.sqrt(n_bootstrap_replications)))
```

Comment:
This is a constant - so you probably don't need to recompute it every call. You could potentially make these module "settings" variables, e.g.

```python
N_BOOTSTRAP_REPLICATIONS = 30
BS_MEAN = 0
...

def _test_bootstrap_samples_from_eval_distribution(random_state):
 ...
```

Reply:
I think I'll leave this where it is for now, more or less. Agreed that having test settings/a config would be useful. I'll probably end up splitting this test up into multiple sub-components that I can re-use for e.g. numerical tests (e.g. "run this test n times and fail if it fails more than once"); I'll probably end up parametrising this particular function over the evaluation distribution, so I'll leave it for the next round of tests for now.